src/librustc_metadata/encoder.rs

```diff
@@ -1699,20 +1678,17 @@ impl<'a, 'b, 'tcx> IndexBuilder<'a, 'b, 'tcx> {
 
     fn encode_info_for_generics(&mut self, generics: &hir::Generics) {
         for param in &generics.params {
-            match param.kind {
-                hir::GenericParamKind::Lifetime { .. } => {}
-                hir::GenericParamKind::Type { ref default, .. } => {
-                    let def_id = self.tcx.hir().local_def_id(param.id);
-                    let has_default = Untracked(default.is_some());
-                    let encode_info = IsolatedEncoder::encode_info_for_ty_param;
-                    self.record(def_id, encode_info, (def_id, has_default));
-                }
-                hir::GenericParamKind::Const { .. } => {
-                    let def_id = self.tcx.hir().local_def_id(param.id);
-                    let encode_info = IsolatedEncoder::encode_info_for_const_param;
-                    self.record(def_id, encode_info, def_id);
-                }
-            }
+            let encode_type = match param.kind {
+                GenericParamKind::Lifetime { .. } => continue,
```

Comment:
Is this the right way to go about this? I guess `ty::Generics` are encoded as part of the generic item, and this part just handles extra information?

Does anything break if you only call `encode_info_for_generic_param` when `encode_type` is `true`?

Reply:
> I guess `ty::Generics` are encoded as part of the generic item, and this part just handles extra information?

This part just handles type information.

> Does anything break if you only call `encode_info_for_generic_param` when `encode_type` is `true`?

Let's find out...

Reply:
> Does anything break
[Yes](https://github.com/rust-lang/rust/pull/58581#issuecomment-467643962).

Reply:
But if you want to resolve this quickly, r=me and open an issue about generic parameters.